apps/create-plugin/src/index.ts

```diff
@@ -0,0 +1,141 @@
+import { Command } from 'commander'
+import packageJson from '../package.json'
+import { createPlugin, logBoostStars } from './builder'
+const _prompts = require('prompts')
+const figlet = require('figlet')
+// import { Chains } from '@rabbitholegg/questdk-plugin-utils'
+import { ActionType } from '@rabbitholegg/questdk'
+
+export enum Chains {
+  ETHEREUM = 1,
+  OPTIMISM = 10,
+  BINANCE_SMART_CHAIN = 56,
+  GNOSIS = 100,
+  POLYGON_POS = 137,
+  ZK_SYNC_ERA = 324,
+  POLYGON_ZK = 1101,
+  MANTLE = 5000,
+  BASE = 8453,
+  ARBITRUM_ONE = 42161,
+  AVALANCHE = 43114,
+  LINEA = 59144,
+  SCROLL = 534352,
+  ZORA = 7777777,
+}
+
+// structure available chains into the format for prompts
+const _chainValues = Object.values(Chains)
+const _chainKeys = Object.keys(Chains)
+const _chainArray = _chainValues
+  .filter((value) => typeof value === 'string')
+  .map((value, _index) => {
+    return { value: value, title: value }
+  })
+
+// structure available actiontypes into the format for prompts
+// and remove Deposit, Lend, and Other which are not currently supported
+const _actionValues = Object.values(ActionType)
+const _actionKeys = Object.keys(ActionType)
+const _actionArray = _actionValues
+  .filter((value) => typeof value === 'string')
+  .map((value, index) => {
+    return { value: _actionKeys[index], title: value }
+  })
+  .filter((value) => value.title !== 'deposit')
+  .filter((value) => value.title !== 'lend')
+  .filter((value) => value.title !== 'other')
+
+// TODO if/when our state gets more complicated we might want to use conf or something like that
+const _pluginName = ''
+
+const _program = new Command()
+_program
+  .version(packageJson.version)
+  .description('A CLI to create a new plugin for the Boost ecosystem')
+  .option('-n, --name  <value>', 'The name of the protocol or platform')
+  .option('-c, --chain  <value>', 'the name of the blockchain')
+  .option('-i, --id  <value>', 'the chain id')
+  .option(
+    '-t, --tx <value>',
+    'An example transaction hash of the action you want to create a plugin for',
+  )
+  .option(
+    '-a, --action <value>',
+    'List of actions to create a plugin for, this is used in the UI',
+  )
+  .parse(process.argv)
+
+const _options = _program.opts()
+
+const _questions = [
+  {
+    type: 'text',
+    name: 'name',
+    message: 'What is the name of the project you are creating a plugin for?',
+    validate: (name: string) => {
+      if (name.length < 1) {
+        return 'Please enter a name'
+      }
+      const invalidNameRegex = new RegExp('[~"#%&*:<>?/\\{|}]+')
```

Comment:
I don't think this validation is tight enough - we could do something like "This is a BIG BAD PROJECT NAME aahahahahahaha" and that would be less than ideal. I think we want something more like this:
```suggestion
      const invalidNameRegex = new RegExp('"^(?:(?:@(?:[a-z0-9-*~][a-z0-9-*._~]*)?/[a-z0-9-._~])|[a-z0-9-~])[a-z0-9-._~]*$"')
```

That's just the package name validation regex, it might not be tight enough but at least it'll prevent breakage.